source/common/upstream/eds_subscription_factory.cc

```diff
@@ -0,0 +1,57 @@
+#include "common/upstream/eds_subscription_factory.h"
+
+namespace Envoy {
+namespace Upstream {
+Config::GrpcMux& EdsSubscriptionFactory::getOrCreateMux(
+    const LocalInfo::LocalInfo& local_info, Grpc::AsyncClientPtr async_client,
+    Event::Dispatcher& dispatcher, const Protobuf::MethodDescriptor& service_method,
+    Runtime::RandomGenerator& random, const ::envoy::api::v2::core::ApiConfigSource& config_source,
+    Stats::Scope& scope, const Config::RateLimitSettings& rate_limit_settings) {
+
+  const uint64_t mux_key = MessageUtil::hash(config_source.grpc_services(0));
```

Comment:
Nit: remove blank lines above/beyond, Google C++ style encourages dense packing of lines except where needed to logically divide up sections.